src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -561,7 +569,7 @@ DeferErrorIfUnsupportedSubqueryPushdown(Query *originalQuery,
 	else if (!RestrictionEquivalenceForPartitionKeys(plannerRestrictionContext))
 	{
 		return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
-							 "cannot pushdown the subquery since all relations are not "
+							 "cannot pushdown the query since all relations are not "
 							 "joined using distribution keys",
```

Comment:
Maybe we should rephrase this message entirely, "all relations are not joined using distribution keys" is also grammatically incorrect. 

Maybe:

```
ERROR: complex joins are only supported when all distributed tables are joined on their partition columns
```